common/src/main/java/com/revenuecat/purchases/common/telemetry/TelemetryManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.revenuecat.purchases.common.telemetry
+
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.verboseLog
+
+class TelemetryManager(
+    private val telemetryFileHelper: TelemetryFileHelper,
+    private val telemetryAnonymizer: TelemetryAnonymizer,
+    private val backend: Backend,
+    private val telemetryDispatcher: Dispatcher,
+    private val telemetryEnabled: Boolean
+) {
+    fun syncTelemetryFileIfNeeded() {
+        if (!telemetryEnabled) {
+            verboseLog("Telemetry disabled. Skipping syncing events.")
+            return
+        }
+        telemetryDispatcher.enqueue(
+            command = {
+                if (telemetryFileHelper.telemetryFileIsEmpty()) return@enqueue
+                val telemetryList = telemetryFileHelper.readTelemetryFile()
+                val telemetryCount = telemetryList.size
+                backend.postTelemetry(
+                    telemetryList = telemetryList,
+                    onSuccessHandler = {
+                        verboseLog("Synced telemetry file successfully.")
+                        telemetryFileHelper.cleanSentTelemetry(telemetryCount)
```

Comment:
This is so if we track an event while there is a sync in progress, we still send it on the next configure call.